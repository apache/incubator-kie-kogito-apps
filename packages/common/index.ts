export {
  default as PageToolbar
} from './src/components/Molecules/PageToolbar/PageToolbar';
export {
  default as PageLayout
} from './src/components/Templates/PageLayout/PageLayout';
export {
  default as AboutModalBox
} from './src/components/Molecules/AboutModalBox/AboutModalBox';
export {
  default as BrandLogo
} from './src/components/Atoms/BrandLogo/BrandLogo';
export {
  default as PageNotFound
} from './src/components/Molecules/PageNotFound/PageNotFound';
export {
  default as ServerUnavailable
} from './src/components/Molecules/ServerUnavailable/ServerUnavailable';
export { default as NoData } from './src/components/Molecules/NoData/NoData';
export {
  default as ServerErrors
} from './src/components/Molecules/ServerErrors/ServerErrors';
export {
  default as ProcessDescriptor
} from './src/components/Molecules/ProcessDescriptor/ProcessDescriptor';
export {
  default as KogitoEmptyState
} from './src/components/Atoms/KogitoEmptyState/KogitoEmptyState';
export * from './src/utils/OuiaUtils';
export * from './src/graphql/types';
