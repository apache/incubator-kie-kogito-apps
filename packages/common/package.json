{
  "name": "@kogito-apps/common",
  "version": "1.0.0",
  "private": true,
  "main": "index.js",
  "author": "",
  "license": "Apache-2.0",
  "repository": {
    "type": "git",
    "url": "https://github.com/kiegroup/kogito-apps.git"
  },
  "scripts": {
    "precommit": "lint-staged",
    "build:prod": "yarn run lint && yarn run test",
    "test:report": "yarn test --ci --reporters=jest-junit",
    "test": "jest --runInBand --ci --reporters=default --reporters=jest-junit",
    "update-snapshot": "jest --updateSnapshot",
    "test:coverage": "jest --coverage",
    "lint": "tslint -c ./tslint.json --project . './src/**/*.ts{,x}'",
    "format": "prettier --config ../../.prettierrc --check --write './src/**/*.{tsx,ts,js}'",
    "clean": "rimraf dist",
    "codegen": "graphql-codegen"
  },
  "dependencies": {},
  "jest": {
    "preset": "ts-jest/presets/js-with-ts",
    "setupFiles": [
      "../../config/Jest-config/test-shim.js",
      "../../config/Jest-config/test-setup.js",
      "core-js"
    ],
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js"
    ],
    "snapshotSerializers": [
      "enzyme-to-json/serializer"
    ],
    "transform": {
      "^.+\\.(ts|tsx)$": "ts-jest",
      "\\.(jpg|jpeg|png|svg)$": "../../config/Jest-config/fileMocks.js"
    },
    "testMatch": [
      "**/tests/*.(ts|tsx|js)"
    ],
    "moduleNameMapper": {
      "\\.(scss|sass|css)$": "identity-obj-proxy"
    }
  },
  "lint-staged": {
    "*.{ts,tsx}": [
      "yarn run format",
      "yarn run lint"
    ]
  }
}
