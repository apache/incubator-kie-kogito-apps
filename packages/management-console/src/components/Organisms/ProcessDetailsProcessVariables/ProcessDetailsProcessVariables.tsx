import {
  Card,
  CardBody,
  CardHeader,
  TextContent,
  Title
} from '@patternfly/react-core';
import React from 'react';
import ReactJson from 'react-json-view';

const ProcessDetailsProcessVariables = ({ data }) => {
  return (
    <Card>
      <CardHeader>
        <Title headingLevel="h3" size="xl">
          Process Variables
        </Title>
      </CardHeader>
      <CardBody>
        <TextContent>
          {data.ProcessInstances.map((item, index) => {
            return (
              <div key={index}>
                <ReactJson src={JSON.parse(item.variables)} />
              </div>
            );
          })}
        </TextContent>
      </CardBody>
    </Card>
  );
};

export default ProcessDetailsProcessVariables;
