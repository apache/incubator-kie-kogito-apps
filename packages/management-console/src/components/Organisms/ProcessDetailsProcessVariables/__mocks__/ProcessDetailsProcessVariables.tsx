import React from 'react'

const MockedProcessVariables = (): React.ReactElement => {
    return <></>
}

export default MockedProcessVariables