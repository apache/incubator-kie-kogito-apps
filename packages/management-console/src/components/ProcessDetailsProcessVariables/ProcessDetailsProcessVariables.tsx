import React from 'react';
import { Card, CardHeader, CardBody, CardFooter, Text, TextVariants, TextContent } from '@patternfly/react-core';
import ReactJson from 'react-json-view';
import JSONViewer from 'react-json-viewer';

const ProcessDetailsProcessVariables = ({ loading, data }) => {
  return (
    <Card style={{ marginTop: '2em', overflowX: 'scroll' }}>
      <CardHeader>Process Variables</CardHeader>
      <CardBody>
        <TextContent style={{ width: '30em' }}>
          {!loading ? (
            data.ProcessId.map((item, index) => {
              return (
                <div key={index}>
                  <ReactJson src={JSON.parse(item.variables)}></ReactJson>
                  <br />
                </div>
              );
            })
          ) : (
            <Text component={TextVariants.h4}>Loading...</Text>
          )}
        </TextContent>
      </CardBody>
      <CardFooter></CardFooter>
    </Card>
  );
};

export default ProcessDetailsProcessVariables;
