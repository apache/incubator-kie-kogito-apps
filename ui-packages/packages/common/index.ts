export {
  default as DataTable
} from './src/components/Organisms/DataTable/DataTable';
export {
  default as KogitoSpinner
} from './src/components/Atoms/KogitoSpinner/KogitoSpinner';
export {
  default as PageToolbar
} from './src/components/Molecules/PageToolbar/PageToolbar';
export {
  default as KogitoPageLayout
} from './src/components/Templates/KogitoPageLayout/KogitoPageLayout';
export {
  default as AboutModalBox
} from './src/components/Molecules/AboutModalBox/AboutModalBox';
export {
  default as EndpointLink
} from './src/components/Atoms/EndpointLink/EndpointLink';
export {
  default as PageNotFound
} from './src/components/Molecules/PageNotFound/PageNotFound';
export {
  default as ServerUnavailable
} from './src/components/Molecules/ServerUnavailable/ServerUnavailable';
export { default as NoData } from './src/components/Molecules/NoData/NoData';
export {
  default as ServerErrors
} from './src/components/Molecules/ServerErrors/ServerErrors';
export {
  default as ProcessDescriptor
} from './src/components/Molecules/ProcessDescriptor/ProcessDescriptor';
export * from './src/components/Atoms/KogitoEmptyState/KogitoEmptyState';
export { default as LoadMore } from './src/components/Atoms/LoadMore/LoadMore';
export {
  default as DomainExplorer
} from './src/components/Organisms/DomainExplorer/DomainExplorer';
export {
  default as DomainExplorerListDomains
} from './src/components/Organisms/DomainExplorerListDomains/DomainExplorerListDomains';
export * from './src/utils/OuiaUtils';
export * from './src/graphql/types';
