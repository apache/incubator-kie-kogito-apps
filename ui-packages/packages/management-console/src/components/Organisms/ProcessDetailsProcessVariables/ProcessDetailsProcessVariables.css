.kogito-management-console--variables__label-fadeOut {
    opacity:0;
     width:0;
     height:0;
     transition: width 0.5s 0.5s, height 0.5s 0.5s, opacity 0.5s;
}

.kogito-management-console--variables__label-fadeIn{
    opacity:1;
     transition: width 0.5s, height 0.5s, opacity 0.5s 0.5s;
}