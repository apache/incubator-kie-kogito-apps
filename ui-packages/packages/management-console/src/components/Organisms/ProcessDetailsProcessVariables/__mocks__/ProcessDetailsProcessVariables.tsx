import React from 'react';

const MockedProcessDetailsProcessVariables = (): React.ReactElement => {
  return <></>;
};

export default MockedProcessDetailsProcessVariables;
