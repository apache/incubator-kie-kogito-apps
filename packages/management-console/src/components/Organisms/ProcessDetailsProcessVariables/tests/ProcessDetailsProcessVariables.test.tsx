import React from 'react';
import ProcessDetailsProcessVariables from '../ProcessDetailsProcessVariables';
import { getWrapper } from '@kogito-apps/common';

jest.mock('react-json-view', () => {
  const MockedJsonDetailsComponent = () => <div />;
  return MockedJsonDetailsComponent;
})

const props = {
  loading: true,
  data: {
    ProcessInstances: []
  }
};

const props2 = {
  loading: false,
  data: {
    ProcessInstances: [
      {
        variables:
          '{"trip":{"begin":"2019-10-22T22:00:00Z[UTC]","city":"Berlin","country":"Germany","end":"2019-10-30T22:00:00Z[UTC]","visaRequired":false},"hotel":{"address":{"city":"Berlin","country":"Germany","street":"street","zipCode":"12345"},"bookingNumber":"XX-012345","name":"Perfect hotel","phone":"09876543"},"traveller":{"address":{"city":"Karkow","country":"Poland","street":"palna","zipCode":"200300"},"email":"rob@redhat.com","firstName":"Rob","lastName":"Rob","nationality":"Polish"}}'
      }
    ]
  }
};
describe('Process Variables component', () => {
  it('Sample test case', () => {
    const wrapper = getWrapper(<ProcessDetailsProcessVariables {...props}/>, 'ProcessDetailsProcessVariables');
    expect(wrapper).toMatchSnapshot();
  });
  it('Assertion for props', () => {
    const wrapper = getWrapper(<ProcessDetailsProcessVariables {...props2}/>, 'ProcessDetailsProcessVariables');
    expect(wrapper).toMatchSnapshot();
  });
});
