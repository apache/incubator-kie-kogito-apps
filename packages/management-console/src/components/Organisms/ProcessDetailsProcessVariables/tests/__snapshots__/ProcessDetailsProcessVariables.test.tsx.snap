// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProcessVariables component tests snapshot testing with variables 1`] = `
<Card>
  <CardHeader>
    <Title
      headingLevel="h3"
      size="xl"
    >
      Process Variables
    </Title>
  </CardHeader>
  <CardBody>
    <TextContent>
      <div
        key="0"
      >
        <t
          collapseStringsAfterLength={false}
          collapsed={false}
          defaultValue={null}
          displayDataTypes={true}
          displayObjectSize={true}
          enableClipboard={true}
          groupArraysAfterLength={100}
          iconStyle="triangle"
          indentWidth={4}
          name="root"
          onAdd={false}
          onDelete={false}
          onEdit={false}
          onSelect={false}
          shouldCollapse={false}
          sortKeys={false}
          src={
            Object {
              "hotel": Object {
                "address": Object {
                  "city": "Berlin",
                  "country": "Germany",
                  "street": "street",
                  "zipCode": "12345",
                },
                "bookingNumber": "XX-012345",
                "name": "Perfect hotel",
                "phone": "09876543",
              },
              "traveller": Object {
                "address": Object {
                  "city": "Karkow",
                  "country": "Poland",
                  "street": "palna",
                  "zipCode": "200300",
                },
                "email": "rob@redhat.com",
                "firstName": "Rob",
                "lastName": "Rob",
                "nationality": "Polish",
              },
              "trip": Object {
                "begin": "2019-10-22T22:00:00Z[UTC]",
                "city": "Berlin",
                "country": "Germany",
                "end": "2019-10-30T22:00:00Z[UTC]",
                "visaRequired": false,
              },
            }
          }
          style={Object {}}
          theme="rjv-default"
          validationMessage="Validation Error"
        />
      </div>
    </TextContent>
  </CardBody>
</Card>
`;

exports[`ProcessVariables component tests snapshot testing without variables 1`] = `
<Card>
  <CardHeader>
    <Title
      headingLevel="h3"
      size="xl"
    >
      Process Variables
    </Title>
  </CardHeader>
  <CardBody>
    <TextContent />
  </CardBody>
</Card>
`;
